train_dreambooth.py

```diff
@@ -148,32 +166,46 @@ def run(args):
     train_dataset = data_util.prepare_datasets()
 
     print("Initializing trainer...")
-    ckpt_path_prefix = (
-        run_name
-    ) = f"lr@{args.lr}-max_train_steps@{args.max_train_steps}-train_text_encoder@{args.train_text_encoder}"
+    ckpt_path_prefix = run_name
     dreambooth_trainer = prepare_trainer(
         args.img_resolution, args.train_text_encoder, args.mp
     )
-    train(dreambooth_trainer, train_dataset, ckpt_path_prefix, args.max_train_steps)
 
+    callbacks = []
     if args.log_wandb:
-        print("Logging artifacts...")
-        wandb.init(project="dreambooth-keras", name=run_name, config=vars(args))
-        if args.validation_prompt is None:
-            args.validation_prompt = (
-                f"A photo of {args.unique_id} {args.class_category} in a bucket"
+        # log training metrics to Weights & Biases
+        callbacks.append(WandbMetricsLogger(log_freq="batch"))
+        # log model checkpoints to Weights & Biases as artifacts
+        callbacks.append(
+            DreamBoothCheckpointCallback(ckpt_path_prefix, save_weights_only=True)
+        )
+        # perform inference on validation prompts at the end of every epoch and
+        # log the resuts to a Weights & Biases table
+        callbacks.append(
+            QualitativeValidationCallback(
+                img_heigth=args.img_resolution,
+                img_width=args.img_resolution,
+                prompts=validation_prompts,
+                num_imgs_to_gen=args.num_images_to_generate,
             )
+        )
+    else:
+        callbacks.append(
+            tf.keras.callbacks.ModelCheckpoint(ckpt_path_prefix, save_weights_only=True)
+        )
 
+    train(dreambooth_trainer, train_dataset, args.max_train_steps, callbacks)
+
+    if args.log_wandb:
```

Comment:
nit: maybe it is better to reuse `StableDiffusion` from `QualitativeValidationCallback` since it is always there under `args.log_wandb=True` condition.